tests/e2e/cypress/e2e/page-load-speed.cy.js

```diff
@@ -0,0 +1,25 @@
+describe("load pages", () => {
+    it("is not a test, is just a utility", () => {
+      const pages = [
+        { name: "location page 1", url: "/local/OKX/33/35/Hoboken" },
+        { name: "location page 2", url: "/local/LOX/155/45/Vernon" },
+        { name: "location page 3", url: "/local/LOT/75/73/Chicago" },
+        { name: "location page 4", url: "/local/HGX/65/97/Houston" },
+        { name: "location page 5", url: "/local/PSR/159/58/Guadalupe" },
+      ];
+      let totalTime = 0;
+  
+      for (const { name, url } of pages) {
+        const startTime = performance.now();
+
+        cy.visit(url);
+        cy.wrap(performance.now()).then(endTime => {
+            totalTime += startTime - endTime;
+            cy.log(`${name}: page load took ${startTime - endTime} milliseconds.`);
+        })
+      }
```

Comment:
What makes the timing measurements tricky is that Cypress is asynchronous, but it tries to look synchronous. I guess they wanted it to be easy to think about, and most of the time it is, but when you need to measure times, it can be tough.

So for example, what's happening in this loop is that we take a timestamp, then enqueue a page visit, then take another timestamp, then enqueue a handler for when Cypress finishes the page "visit" using the second timestamp as the end time. What you've done is pretty clever using `wrap()`. Unfortunately, it's capturing a timestamp that happens immediately after the page visit is enqueued. It may or may not cover the actual time needed to visit the page. (Cypress will do it whenever it gets around to it; we don't control it.)

What I was seeing was that the start and end times for all of the pages were very nearly the same, which makes sense because they are being measured inside the loop, and the loop isn't blocking so the reported page load times are too short. Our page loads are longer than the tests are saying.

Something like this will work:

```js
const startTime = performance.now();

cy.visit(url).then(() => {
  const endTime = performance.now();
  totalTime += startTime - endTime;
  cy.log(`${name}: page load took ${startTime - endTime} milliseconds.`);
});
```

The important thing is not to call `performance.now()` until after Cypress has completed the page visit.